static/app/views/profiling/landing/profileCharts.tsx

```diff
@@ -0,0 +1,145 @@
+import {useMemo} from 'react';
+import {InjectedRouter} from 'react-router';
+import {useTheme} from '@emotion/react';
+
+import {AreaChart} from 'sentry/components/charts/areaChart';
+import ChartZoom from 'sentry/components/charts/chartZoom';
+import {Panel} from 'sentry/components/panels';
+import {PageFilters} from 'sentry/types';
+import {Series} from 'sentry/types/echarts';
+import {axisLabelFormatter, tooltipFormatter} from 'sentry/utils/discover/charts';
+import {useProfileStats} from 'sentry/utils/profiling/hooks/useProfileStats';
+
+interface ProfileChartsProps {
+  query: string;
+  router: InjectedRouter;
+  selection?: PageFilters;
+}
+
+// We want p99 to be before p75 because echarts renders the series in order.
+// So if p75 is before p99, p99 will be rendered on top of p75 which will
+// cover it up.
+const SERIES_ORDER = ['count()', 'p99()', 'p75()'];
+
+export function ProfileCharts({query, router, selection}: ProfileChartsProps) {
+  const theme = useTheme();
+
+  const profileStats = useProfileStats({query, selection});
+  const series: Series[] = useMemo(() => {
+    if (profileStats.type !== 'resolved') {
+      return [];
+    }
+
+    const timestamps = profileStats.data.timestamps;
+
+    const allSeries = profileStats.data.data.map(rawData => {
+      if (timestamps.length !== rawData.values.length) {
+        throw new Error('Invalid stats response');
+      }
+
+      if (rawData.axis === 'count') {
+        return {
+          data: rawData.values.map((value, i) => ({
+            name: timestamps[i] * 1e3,
+            value: value ?? 0,
```

Comment:
What is the default chart behavior here? Like if the value is undefined, does it skip drawing for that interval and we set it to 0 to prevent that?

Reply:
The default chart behaviour here is to skip that point and connect the non-null values, this prevents it by using 0s.